snakemake/parser.py

```diff
@@ -804,7 +804,7 @@ def __exit__(self, *args):
 def format_tokens(tokens):
     t_ = None
     for t in tokens:
-        if t_ and not t.isspace() and not t_.isspace():
+        if t_ and (not t.isspace() and not t_.isspace()) and (not t_ == '@'):
```

Comment:
This change is needed to treat decorator correctly.
Before this change, parser changes "@decorator" to "@ decorator", and Python interpreter does not treat it as decorator.